doc/guides/4 - Refinery Extensions/7 - changes to extensions in Refinery 3.0.textile

```diff
@@ -0,0 +1,194 @@
+
+<pre>
+<code>
+// refinerycms-blog/app/assets/stylesheets/blog/refinery/backend.css.scss
+.comments_icon {
+  @include icon('comments')
+}
+.comment_icon {
+  @include icon('comment')
+}
+.comment_new_icon {
+  @include icon('comment-o')
+}
+
+// previously .comment_cross_icon
+.comment_rejected_icon {
+  @include icon('thumbs-o-down', red)
+}
+
+//previously .comment_tick_icon
+.comment_approved_icon {
+  @include icon('thumbs-o-up', green)
+}
+</code>
+</pre>
+
+To use the @icon@ mixin we need to import it from Refinery::Core. Also Rails is phasing out the use of *.css.scss in favour of simple *.scss files.
+
+So I renamed backend.css.scss to backend.scss and imported some mixins at hte top of the file.
+
+<pre>
+<code>
+// refinerycms-blog/app/assets/stylesheets/blog/refinery/backend.css.scss
+
+@import 'refinery/mixins';
+@import 'refinery/colours';
+@import 'font-awesome-sprockets', 'font-awesome';
+@import 'refinery/icons';
+
+.comments_icon {
+  @include icon('comments')
+}
+...
+</code>
+</pre>
+
+
+
+h2. Assets
+
+To add your assets to the assets defined with Refinery use @config/initializers/assets.rb@. A simple assets.rb would look like this:
+
+<pre>
+<code>
+  Refinery::Core.configure do |config|
+    # Register extra javascripts for backend
+    config.register_javascript "refinerycms-testimonials.js"
+  
+    # Register extra stylesheets for backend 
+    config.register_stylesheet "refinerycms-testimonials", :media => 'screen'
+  end
+  Rails.application.config.assets.precompile += %w( refinerycms-testimonials.css refinerycms-testimonials.js )
+</code>
+</pre>
+
+Using "namespaced" filenames, as above, keeps your code linked to your extension and also helps to keep your stylesheets and javascripts from being overridden accidentally by users. 
+
+h2. Decorators and Strong Parameters
+
+Rails 4 requires the use of strong parameters. (You can turn them off altogether but that is not recommended)
+
+If your engine adds fields to a page, blog post or other defined object you need to add your fields to the page's list of acceptable parameters. Here is how to do that.
+
+<pre>
+<code>
+
+Refinery::Admin::PagesController.class_eval do
+
+  # work around an issue with stack level too deep, due to an issue with decorators.
+  if self.instance_methods.exclude?(:page_params_with_testimonial_params)
+    def page_params_with_testimonial_params
+
+      testimonial_params = params.require(:page).permit(:testimonials_show, :testimonials_count, :testimonials_select)
+      page_params_without_testimonial_params.merge(testimonial_params)
+
+    end
+
+    # Swap out the default page_params method with our new one
+    alias_method_chain :page_params, :testimonial_params
+  end
+end
```

Comment:
Can we use this approach instead:

```ruby
module PageParamsWithTestimonialParamsMixin
  def permitted_page_params
    super.merge(:testimonials_show, :testimonials_count, :testimonials_select)
  end
end

Refinery::Admin::PagesController.class_eval do
  prepend PageParamsWithTestimonialParamsMixin
end
```

This is due to the method on that controller which was added for this purpose.